### PR TITLE
Added NASA Profile

### PR DESCRIPTION
**Pull request by anjomro**
Hey,
I added support for the NASA ( Nahverkehr Sachsen-Anhalt). Unfortunately the format of the HAFAS-API is a bit different than teh default so I had to override the `Journey` and the `Journeys` classes for the profile.
I also added some tests, at least for the requests and docs follwoing the existing docs.

In my tests the API of NASA seemed to work fine, although sometimes the request randomly failed and succeded on the next run.

I would be glad if my PR could be merged, otherwise I would appreciate some comments what needs to be improved.

Best Regards, Anton!

### CODE REVIEW

**Comment by 1Maxnet1** [comment]
@anjomro You want to resolve the conflicts with the main branch of this repository?

**Comment by 1Maxnet1** [comment]
> Hi @1Maxnet1, Honestly I thought thought this repo was dead, since apparently no one had a look at my PR for one and a half years.

Yeah understandable. But as we managed to have some recent activity again (see 0.4.0 release) I thought I might ask :)

**Comment by leona-ya** [comment]
> I hope this PR can be merged soon?

will review later when i'm in a train – do train software in trains ;)

Sorry that it took so long to merge this. The reason is a mix of me having way too much to do over the last years, and that I also thought this project was dead. This project came about because we needed a hafas client for our own software. Then this use-case died down a bit, and it seemed to me that nobody else was interested in pyhafas. Now it's slowly coming back to life, and I'm happy about that.

**Comment by anjomro** [comment]
> will review later when i'm in a train – do train software in trains ;)

A very useful approach for managing time: Home Automation at Home, Work Software at work and train software in trains :D

> Sorry that it took so long to merge this. The reason is a mix of me having way too much to do over the last years, and that I also thought this project was dead. This project came about because we needed a hafas client for our own software. Then this use-case died down a bit, and it seemed to me that nobody else was interested in pyhafas. Now it's slowly coming back to life, and I'm happy about that.

I totally understand you, lets hope that it will take on some speed again. Although you might need to spend a lot more time in trains in that case :grin:

File: tests/nasa/requests/journey_test.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from pyhafas import HafasClient
+from pyhafas.profile import NASAProfile
+from pyhafas.types.fptf import Journey
+
+
+# Test journey request with NASA profile
+def test_nasa_journey_request():
+    client = HafasClient(NASAProfile())
+    journey = client.journey(
+        journey="¶HKI¶T$A=1@O=Magdeburg, Universitätsbibliothek@L=7482@a=128@$A=1@O=Magdeburg, Neue Str./Zirkusmuseum@L=7411@a=128@$202202271954$202202272011$Str    2$$1$$$$¶KCC¶I1ZFIzEjRVJHIzEjSElOIzAjRUNLIzE4NDc0fDE4NDc0fDE4NDkxfDE4NDkxfDB8MHw1fDE4NDYxfDF8LTIxNDc0ODM2NDZ8MCNHQU0jMjcwMjIyMTk1NCMKWiNWTiMxI1NUIzE2NDU3ODQ5NjEjUEkjMCNaSSMyMzg3I1RBIzQjREEjMjcwMjIyIzFTIzc0NzgjMVQjMTk0OCNMUyM3NTQwI0xUIzIwMjYjUFUjODAjUlQjMSNDQSNTdE4jWkUjMiNaQiNTdHIgICAgMiNQQyM1I0ZSIzc0ODIjRlQjMTk1NCNUTyM3NDExI1RUIzIwMTEj",
```
Comment (leona-ya) [inline tests/nasa/requests/journey_test.py:10]: For this to work we need to have a newer journey. I would suggest 
```¶HKI¶T$A=1@O=Magdeburg Hbf@L=8010224@a=128@$A=1@O=Halle (Saale) Hbf@L=8010159@a=128@$202307071735$202307071841$    RE30$$1$$$$$$¶KC¶#VE#2#CF#100#CA#0#CM#0#SICT#0#AM#49#AM2#0#RT#15#¶KCC¶I1ZFIzEjRVJHIzEjSElOIzAjRUNLIzE0MDE1fDE0MDE1fDE0MDgxfDE0MDgxfDB8MHw1fDE0MDEwfDF8MHwyfDB8MHwtMjE0NzQ4MzY0OCNHQU0jNzA3MjMxNzM1IwpaI1ZOIzEjU1QjMTY4ODcxMzcxNSNQSSMwI1pJIzYyNTA3I1RBIzAjREEjNzA3MjMjMVMjODAxMDIyNCMxVCMxNzM1I0xTIzgwMTAxNTkjTFQjMTg0MSNQVSM4MCNSVCMxI0NBI1JFI1pFI1JFMzAjWkIjICAgIFJFMzAjUEMjMyNGUiM4MDEwMjI0I0ZUIzE3MzUjVE8jODAxMDE1OSNUVCMxODQxIw==¶KRCC¶#VE#1#```
(i can also change this, if you want).

**Review by leona-ya — COMMENTED** [review]
Thank you very much! It seems good, just one small note to fix the one test. After that I'm happy to merge this

File: tests/nasa/requests/journey_test.py (hunk 1 above)
Comment (anjomro) [reply tests/nasa/requests/journey_test.py]: I updated the journey. Apparently they now use a different format for the station id now. Previously it was just the four digit id like **7482**, now its more complex like: 300**7482**02.

**Review by leona-ya — APPROVED** [review]

**Pull request merged**